File: 776-n-ary-tree-postorder-traversal/n-ary-tree-postorder-traversal.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def postorder(self, root: 'Node') -> List[int]:
        a=[]
        def dfs(cur):
            if cur!=None:
                for i in cur.children:
                    dfs(i)
                a.append(cur.val)
        dfs(root)
        return a

File: 776-n-ary-tree-postorder-traversal/n-ary-tree-postorder-traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    vector<int> dfs(Node* cur,vector<int>& a){
        if (cur!=NULL){
            for(auto i:cur->children){
                dfs(i,a);
            }
            a.push_back(cur->val);
        }
        return a;
    }
public:
    vector<int> postorder(Node* root) {
        vector<int> a;
        dfs(root,a);
        return a;
    }
};